tajo-core/src/main/java/org/apache/tajo/session/Session.java

```diff
@@ -121,6 +124,10 @@ public synchronized String getCurrentDatabase() {
     return currentDatabase;
   }
 
+  public synchronized void setQueryCache(LoadingCache<String, Expr> cache) { this.cache = cache; }
```

Comment:
Currently, in this patch, I try to minimize dependency change in order to keep Tajo's good modularity. So, I put only Expr dependency into Session class. (It's inevitable.) If we put the cache initialization there, we need "import org.apache.tajo.engine.parser.SQLAnalyzer" there. I am not sure that Tajo allows this kind of dependency. If you think this is okay, I prefer to move that into Session. I need your guide. :)